nixos/modules/services/web-apps/peertube.nix

```diff
@@ -0,0 +1,186 @@
+      '';
+    };
+
+    database = {
+      createLocally = lib.mkOption {
+        description = "Configure local PostgreSQL database server for PeerTube.";
+        type = lib.types.bool;
+        default = true;
+      };
+
+      name = lib.mkOption {
+        type = lib.types.str;
+        default = "peertube_prod";
+        description = "Database name.";
+      };
+
+      user = lib.mkOption {
+        type = lib.types.str;
+        default = "peertube";
+        description = "Database user.";
+      };
+    };
+
+    smtp = {
+      createLocally = lib.mkOption {
+        description = "Configure local Postfix SMTP server for PeerTube.";
+        type = lib.types.bool;
+        default = true;
+      };
+    };
+
+    redis = {
+      createLocally = lib.mkOption {
+        description = "Configure local Redis server for PeerTube.";
+        type = lib.types.bool;
+        default = true;
+      };
+    };
+
+    runtimeDir = lib.mkOption {
+      type = lib.types.path;
+      default = "/var/lib/${name}";
+      description = "The directory where Peertube stores its runtime data.";
+    };
+
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = pkgs.peertube;
+      description = ''
+        Peertube package to use.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    users.users = lib.optionalAttrs (cfg.user == name) {
+      "${name}" = {
+        inherit uid;
+        group = cfg.group;
+        description = "Peertube user";
+        home = cfg.runtimeDir;
+        useDefaultShell = true;
+        # todo: fix this. needed for postgres authentication
+        password = "peertube";
+      };
+    };
+    users.groups = lib.optionalAttrs (cfg.group == name) {
+      "${name}" = {
+        inherit gid;
+      };
+    };
+
+    services.postgresql = lib.mkIf cfg.database.createLocally {
+      enable = true;
+      ensureUsers = [ { name = cfg.database.user; }];
+      # The database is created as a startup script of the peertube service.
+      authentication = ''
+        host ${cfg.database.name} ${cfg.database.user} 127.0.0.1/32 trust
+        host ${cfg.database.name} ${cfg.database.user} 127.0.0.1/32 md5
+      '';
+    };
+
+    services.postfix = lib.mkIf cfg.smtp.createLocally {
+      enable = true;
+    };
+
+    services.redis = lib.mkIf cfg.redis.createLocally {
+      enable = true;
+    };
+
+    # Make sure the runtimeDir exists with the desired permissions.
+    systemd.tmpfiles.rules = [
+      "d \"${cfg.runtimeDir}\" - ${cfg.user} ${cfg.group} - -"
+    ];
+
+    systemd.services.peertube = {
+      description = "Peertube";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" "postgresql.service" "redis.service" ];
+      wants = [ "postgresql.service" "redis.service" ];
+
+      environment.NODE_CONFIG_DIR = "${cfg.runtimeDir}/config";
+      environment.NODE_ENV = "production";
+      environment.HOME = cfg.package;
```

Comment:
```suggestion
      environment.HOME = cfg.package;
      environment.NODE_EXTRA_CA_CERTS = "/etc/ssl/certs/ca-certificates.crt";
```

Reply:
Just an proposal but this make it work for me out of the box with self-signed certificates added to my local trust store.

Reply:
Hmm, where does that magic string `"/etc/ssl/certs/ca-certificates.crt"` come from? Could we perhaps add this conditional on some nixos variable? Or is `"/etc/ssl/certs/ca-certificates.crt"` the standard location always for certs in NixOS?

Reply:
https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/security/ca.nix#L85 it's called the canonical location so should be fine.

Reply:
Also many other packages reference it or the old /etc/ssl/certs/ca-bundle.crt one.